.github/workflows/version-check.yml

```
@@ -10,7 +10,7 @@ permissions:
 
 jobs:
   version-check:
-    if: contains(github.event.pull_request.labels.*.name, 'needs review')
+    if: github.event.pull_request.draft == false
```

Small note:

```suggestion
    if: !github.event.pull_request.draft
```

would read slightly more natural to me and seems to be generally supported by the [Expressions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#operators) of GitHub Workflows.